### PR TITLE
feat: allow changing `iconButtonFactor`

### PR DESCRIPTION
**Pull request by adil192**
# Pull Request

## Description

This PR adds a new field to `QuillToolbarBaseButtonOptions` for the icon button factor, which defaults to the existing `kIconButtonFactor` constant.

The new version of `flutter_quill` has a `kIconButtonFactor` which is inconsistent with the rest of my (non-Quill) buttons, so this PR allows me to change the icon button factor from `1.77` to `1.6`.

## Features

- Add a new field to `QuillToolbarBaseButtonOptions` for the icon button factor, which defaults to the existing `kIconButtonFactor` constant.

## Suggestions
We could also change the default `kIconButtonFactor` to `1.6` which matches other default Flutter buttons better.

## Checklist

- [x] I have added/updated relevant documentation 
- [X] I have tested these changes locally. 
- [X] I have followed the code style and guidelines. 
- [x] I have updated `CHANGELOG.md` with my changes in the next section 
- [x] I have run `dart format .`` on the project 
- [x] I have run `dart fix --apply` on the project 
- [x] I have run `flutter test` and `flutter analyze` and it passed successfully 
- [x] I have run `./before-push.sh` and everything is fine

### CODE REVIEW

**Comment by singerdmx** [comment]
@freshtechtips actually there is one who knows the repo very well

**Comment by freshtechtips** [comment]
> @freshtechtips actually there is one who knows the repo very well 

Glad to hear that. I wasn't expecting a new pull request

File: doc/migration.md [hunk 1]
```diff
@@ -79,10 +79,14 @@ All the options have parent `QuillToolbarBaseButtonOptions` which have common th
   final IconData? iconData;
 
   /// To change the icon size pass a different value, by default will be
-  /// [kDefaultIconSize]
+  /// [kDefaultIconSize].
```
Comment (freshtechtips) [inline doc/migration.md:82]: Not sure if your attention to details can be described using words

**Comment by freshtechtips** [comment]
Just one tiny little thing, did you passed that assign that new parameter to the new created button options in the `childBuilder` on the buttons? because otherwise everyone that is using the `childBuilder` will get invalid options settings

File: doc/migration.md (hunk 1 above)
Comment (adil192) [reply doc/migration.md:82]: I'm sorry, please can you elaborate? I can revert the full-stop if you want me to

Comment (freshtechtips) [reply doc/migration.md:82]: > I'm sorry, please can you elaborate? I can revert the full-stop if you want me to

No need to be sorry. It was a compliment 😐

**Comment by adil192** [comment]
> Just one tiny little thing, did you passed that new parameter to the newly created button options in the `childBuilder` on the buttons? because otherwise everyone that is using the `childBuilder` will get invalid options settings

Good catch, let me add that

**Comment by freshtechtips** [comment]
> > Just one tiny little thing, did you passed that new parameter to the newly created button options in the `childBuilder` on the buttons? because otherwise everyone that is using the `childBuilder` will get invalid options settings
> 
> Good catch, let me add that

You don't have to do it if you want to merge the pull request now,
we can do it later and by later I mean today or tomorrow and not delaying forever
but that is up to you

In both cases we will do it correctly

**Comment by adil192** [comment]
Seeing as tests are passing, I'll remove the dependency_override which will probably make tests fail. This will of course resolve itself when 8.3.0 is published to pub.dev

**Comment by freshtechtips** [comment]
> Seeing as tests are passing, I'll remove the dependency_override which will probably make tests fail. This will of course resolve itself when `flutter_quill: ^8.3.0` and `flutter_quill_extensions: ^0.6.5` are published to pub.dev

Correct, but there is a better way to do it, using pubspec.yaml 

Please take a look at the contributing page

You will find how to use the disable and enable local development 

This way, we don't have to touch the pubspec.yaml in order to test the changes

**Comment by freshtechtips** [comment]
I think we are ready to merge, @singerdmx what do you say?

**Comment by singerdmx** [comment]
LGTM

**Comment by freshtechtips** [comment]
@adil192 Let me know when you are ready to merge

**Pull request merged**